volatility3/cli/volshell/generic.py

```diff
@@ -553,14 +553,9 @@ def create_configurable(
             if argname in kwargs:
                 del kwargs[argname]
 
-        for keyword in kwargs:
-            val = kwargs[keyword]
-            if not isinstance(
-                val, interfaces.configuration.BasicTypes
-            ) and not isinstance(val, list):
-                if not isinstance(val, list) or all(
-                    isinstance(x, interfaces.configuration.BasicTypes) for x in val
-                ):
+        for keyword, val in kwargs.items():
+            if not isinstance(val, (interfaces.configuration.BasicTypes, list)):
+                if all(isinstance(x, interfaces.configuration.BasicTypes) for x in val):
```

Comment:
hm.. it seems the original implementation doesn't support tuples, but yours does, the `if not isinstance(val, list) or` might be intentional.

Anyway, I don't think the original implementation (and your changes) are correct. For instance, if val is of type `float`, this might crash since it’s neither in `BasicTypes`, a list, and it's not iterable. 

@ikelos On the other hand, we need to fix [this](https://github.com/volatilityfoundation/volatility3/blob/b8298a345fc5a3260ca2156d1e81a292530bde6e/volatility3/framework/interfaces/configuration.py#L788), where the values don't align with those in `BasicTypes`. Maybe, that line should use `BasicTypes` and `BasicTypes` needs to be updated too to also include `float` or ... ideally, we should have a dedicated function to centralize type validation for configuration values.

Reply:
So I don't believe floats were supposed to be allowed in config items (because SimpleTypes was supposed to be single point of truth for this), so they can *probably* be removed without problem from the config type verification.  I agree we should unify that list with `BasicTypes`.  I don't think there's necessarily point in a function for verification, but a shared list definitely makes sense (and the code you pointed out should be using it, the only reason I can think why it might legitimately not be using it is if it causes import loops).

Yeah, I've no idea why the ifs seemed to be run twice, but I did write it 7 years ago.  5;P  I'm not sure the replacement works correctly.  It's basically checking that everything's a basic type, or is a list of basic types (so no lists of lists, and no other complex types involved although vol types would be allowed because they all inherit from standard base types).

Reply:
The one you've got (and to be fair I think also the original) allows a list of anything to get through, which wasn't the intention.  The intention was either a `BasicType` or a list where each element is only a `BasicType` (and not a list).

Reply:
This PR became a bit messy (the first commit allows a list of anything to get through), is it better if you close it and I do a new PR with just the change in the last commit?

From the idea of @gcmoreira above, may be good to create a utils.py and choose a location for it. `from volatility3.framework import utils`?